www/source/blog/2018-10-23-habitat-inspec.html.md

```diff
@@ -0,0 +1,236 @@
+
+Yes, there is, and it is awesome. [You can check out the docs here](https://www.inspec.io/docs/reference/habitat/). This integration creates a Habitat package (a HART artifact) which includes the profile itself, the latest version of InSpec, and the Habitat Supervisor.
+
+However, this integration currently does not easily allow integration with Builder as, when run from within the directory of an InSpec profile, it does not create a habitat directory with a plan inside.
+
+The purpose of this post is to show you how you can harness Builder's Github integration feature to automatically trigger new builds for your profiles whenever you commit a change to master. Additionally, you will also be able to take advantage of Builder's dependency management features. If your profile depends on another Builder package and that package is updated, Builder will automatically rebuild your package as well. This is one of the main advantages of Builder.
+
+We could, of course, modify the integration to allow for this - but let's explore a different way for now.
+
+## Pre-requisites
+* A GitHub account
+* Git installed on your workstation
+* An account on [Builder](https://bldr.habitat.sh)
+* An origin on Builder (see [these docs](https://www.habitat.sh/docs/using-builder/) for assistance)
+* Habitat installed on your workstation (see [these docs](https://www.habitat.sh/docs/install-habitat/) for assistance)
+
+## Setup
+
+* Go ahead and fork this [nginx-baseline profile repo](https://github.com/dev-sec/nginx-baseline) to your personal Github account.
+* Clone this repo to your workstation and change directories into it.
+
+```
+$ git clone git@github.com:<YOUR GITHUB NAME>/nginx-baseline.git
+$ cd nginx-baseline
+```
+
+* Now, let's initialize Habitat
+
+```
+$ hab plan init
+(...)
+★ An abode for your code is initialized!
+```
+
+* This will create a "habitat" directory in this repo, along with a plan.sh file. Let's take a look at that file:
+
+**plan.sh**
+```
+pkg_name=nginx-baseline
+pkg_origin=<YOUR ORIGIN>
+pkg_version="0.1.0"
+pkg_scaffolding="core/scaffolding-ruby"
+```
+
+* Because InSpec profiles are written in ruby, Habitat automatically assumed we were creating a plan for a Ruby application and assumed it should use the ruby scaffolding.
+
+* We don't need this scaffold for our purposes today, so let's remove that line
+
+**plan.sh**
+```
+pkg_name=nginx-baseline
+pkg_origin=nshamrell
+pkg_version="0.1.0"
+```
+
+* Then add in this content (note that I've modified the pkg_version string to match the version of nginx-baseline):
+
+**plan.sh**
+```
+pkg_name=nginx-baseline
+pkg_origin=<YOUR ORIGIN>
+pkg_version="2.0.2"
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_license=("Apache-2.0")
+pkg_source=https://github.com/<YOUR GITHUB NAME>/nginx-baseline/archive/2.0.2.tar.gz
+pkg_shasum="b546c3940ce03d72f4bbb77503ecf792fd54e608cd729f6511c03e11c5cd0727"
+pkg_deps=(chef/inspec)
+pkg_bin_dirs=(bin)
+
+do_build() {
+    return 0
```

Comment:
The actual build action for an Inspec profile is `inspec archive [profile]` - that resolves the profile's dependencies and packs them into a tgz

Should that be happening at this point?

Reply:
Oh, I now see at https://www.inspec.io/docs/reference/habitat/ that there is `inspec habitat profile create` which should resolve dependencies and generate a hart. I'm not sure if that hart yields the tgz version of a profile or just packs up all the source files for it